lib/streamlit/elements/metric.py

```diff
@@ -0,0 +1,154 @@
+        """Display a metric in big bold font, with an optional indicator of how the metric changed.
+
+        Parameters
+        ----------
+        label : str
+            The header or Title for the metric
+        value : int, float, str, or None
+             Value of the metric. None is rendered as a long dash.
+        delta : int, float, str, or None
+            Indicator of how the metric changed, rendered with an arrow below
+            the metric. If delta is negative (int/float) or starts with a minus
+            sign (str), the arrow points down and the text is red; else the
+            arrow points up and the text is green. If None (default), no delta
+            indicator is shown.
+        delta_colors : str
+             If "normal" (default), the delta indicator is shown as described
+             above. If "inverse", it is red when positive and green when
+             negative. This is useful when a negative change is considered
+             good, e.g. if cost decreased. If "off", delta is  shown in gray
+             regardless of its value.
+
+        Large values are not automatically shortened in v1 (e.g. 5200 → 5.2k)
+        but we should point developers in the docs to packages like [millify]
+        (https://github.com/azaitsev/millify) or [numerize]
+        (https://github.com/davidsa03/numerize), which can do this.
+
+        -------
+
+        Example
+        -------
+        >>> st.metric(label="Active developers", value=123, delta=123,
+        ...     delta_colors="off")  # arrow up, gray
+
+        """
+        metric_proto = MetricProto()
+        metric_proto.body = self.parse_value(value)
+        metric_proto.label = self.parse_label(label)
+        metric_proto.delta = self.parse_delta(delta)
+
+        color_and_direction = self.determine_delta_color_and_direction(
+            clean_text(delta_colors), delta
+        )
+        metric_proto.color = color_and_direction.color
+        metric_proto.direction = color_and_direction.direction
+
+        return str(self.dg._enqueue("metric", metric_proto))
+
+    def parse_label(self, label):
+        if not isinstance(label, str):
+            raise TypeError(
+                f"'{str(label)}' is not an accepted type. label only accepts: str"
+            )
+        return label
+
+    def parse_value(self, value):
+        if value is None:
+            return "—"
+        if isinstance(value, float) or isinstance(value, int) or isinstance(value, str):
+            return str(value)
+        else:
+            raise TypeError(
+                f"'{str(value)}' is not an accepted type. value only accepts: "
+                "int, float, str, or None"
+            )
+
+    def parse_delta(self, delta):
+        if delta is None:
+            return ""
+        if isinstance(delta, str):
+            delta = dedent(delta)
+            if delta[0] == "+":
+                return delta[1:]
```

Comment:
I think there's a bug here where passing in the empty string will cause `st.metric` to explode, but my guess is that we probably want to treat the empty string the same way as we treat the `delta is None` case here.